llvm/lib/Target/AArch64/AArch64RegisterInfo.td

```diff
@@ -20,33 +20,39 @@ class AArch64Reg<bits<16> enc, string n, list<Register> subregs = [],
 
 let Namespace = "AArch64" in {
   // SubRegIndexes for GPR registers
-  def sub_32 : SubRegIndex<32>;
-  def sube64 : SubRegIndex<64>;
-  def subo64 : SubRegIndex<64>;
-  def sube32 : SubRegIndex<32>;
-  def subo32 : SubRegIndex<32>;
+  def sub_32   : SubRegIndex<32,  0>;
+  def sub_32_hi: SubRegIndex<32, 32>;
```

Comment:
Comment the artificial cases? Could we use an explicit marker for these?

Reply:
I've added a comment to describe the artificial cases. I'm not sure what you mean with an explicit marker though?

Reply:
I mean an isArtificial bit on SubRegIndex

Reply:
`isArtificial` is a property of a Register or a RegisterClass, but not a SubRegIndex.